core/src/test/java/org/apache/spark/memory/TaskMemoryManagerSuite.java

```diff
@@ -25,27 +26,37 @@
 
 public class TaskMemoryManagerSuite {
 
+  static TaskMemoryManager manager;
+
+  @After
+  public void after() {
+    Assert.assertEquals(0, manager.getMemoryConsumptionForThisTask());
+    Assert.assertEquals(0, manager.cleanUpAllAllocatedMemory());
+    manager = null;
+  }
+
   @Test
   public void leakedPageMemoryIsDetected() {
-    final TaskMemoryManager manager = new TaskMemoryManager(
+    manager = new TaskMemoryManager(
       new StaticMemoryManager(
         new SparkConf().set("spark.memory.offHeap.enabled", "false"),
         Long.MAX_VALUE,
         Long.MAX_VALUE,
         1),
       0);
     final MemoryConsumer c = new TestMemoryConsumer(manager);
-    manager.allocatePage(4096, c);  // leak memory
+    final MemoryBlock block = manager.allocatePage(4096, c);  // leak memory
     Assert.assertEquals(4096, manager.getMemoryConsumptionForThisTask());
     Assert.assertEquals(4096, manager.cleanUpAllAllocatedMemory());
+    manager.freePage(block, c);
```

Comment:
It should. We can check for that instead with:
Assert.assertEquals(0, manager.cleanUpAllAllocatedMemory());